server/controllers/userController.js

```diff
@@ -34,40 +35,54 @@ const users = [
   },
 ];
 
-dotenv.config();
 class UserController {
-     static signUp = (req, res) => {
-       const id = users.length + 1;
-       const isEmailTaken = users.find(user => user.email === req.body.email);
-       if (isEmailTaken) {
-         return res.status(HttpStatus.CONFLICT).send({ status: HttpStatus.CONFLICT, error: `${req.body.email} is already taken!` });
-       }
-       let { isMentor } = req.body;
-       if (isMentor === undefined) { isMentor = false; }
-       let isAdmin = false;
+  static model() {
+    return new Model('users');
+  }
+
+  static signUp = async (req, res) => {
+    try {
+      let {
+        firstName,
+        lastName,
+        email,
+        password,
+        address,
+        bio,
+        occupation,
+        expertise,
+      } = req.body;
+      const user = await this.model().select('*', 'email=$1', [email]);
+      if (user[0]) {
+        return res.status(HttpStatus.CONFLICT).json({
```

Comment:
please create a response handler for your responses.. you do not need to repeat res.status().json() all through your codebase

Reply:
yeah already done